src/switch/src/Switch.js

```diff
@@ -65,6 +65,35 @@ CheckIcon.propTypes = {
 
 const noop = () => {}
 
+const pseudoSelectors = {
+  _base: '& + div',
+  _disabled: '&[disabled] + div',
+  _hover: '&:not([disabled]):hover + div',
+  _focus: '&:not([disabled]):focus + div',
```

Comment:
On the parent component we could use `_focusWithin: '&:focus-within'` instead of all this sibling stuff.

We would probably need to bubble the `disabled` attr to the wrapping component, perhaps with `aria` if it makes sense.